packages/accounts/src/msca/builder.ts

```diff
@@ -92,11 +110,39 @@ export class MSCABuilder {
 
   build<TTransport extends SupportedTransports = Transport>(
     params: BaseSmartAccountParams
-  ): MSCA {
+  ): MSCA<TTransport, ReturnType<typeof pluginManagerDecorator>> {
     const builder = this;
     const { signer, executor, factory } = zCompleteBuilder.parse(builder);
 
-    return new (class extends BaseSmartContractAccount<TTransport> {
+    return new (class DynamicMSCA<
+      TProviderDecorators = ReturnType<typeof pluginManagerDecorator>
+    > extends BaseSmartContractAccount<TTransport> {
```

Comment:
nit: does this class definition need to be in-line with a method? can we take it out and put it near the interface and define an instance here (constructor can take in signer, executor, factory)?

Reply:
hmmm yea it could be, but by keeping it here then I can do things that typescript says I can't do because we have the guarantees of this class only ever being defined within the bounds of the builder

I can look into changing it in a follow up PR

Reply:
out of curiosity, what are those things you can only do in this setup (without typescript complaining?) accessing the  signer, executor, factory fields?

Reply:
oh it's not that typescript isn't complaining, it is complaining. But now that I look at it, I think a lot of the issues that I ran into are not here anymore. 

There's a lot of typecasting going on in the anonymous class, but none of it is dependent on the types of the builder (thought at some point it was)

I was thinking about adding a `withProviderExtensions` method to the builder which is where I first ran into angry typescript that was ok to ignore because of the tight coupling. 

I'll take a pass at breaking it out into a concrete class, but what I also wanted to avoid was direct usage of that class in other places. It's really only meant to be used within the builder

Reply:
cool yeah since we're prototyping, let's keep it for now.

fwiw re: only using the class in the builder, we can just not export it, and then any PRs that want to use we can reject